server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitRepositoryManager.java

```diff
@@ -91,6 +97,86 @@ public Project parent() {
+            encryptedRepository = createEncryptedRepository(parent, oldRepository.repoDir(),
+                                                            oldRepository.author(),
+                                                            oldRepository.creationTimeMillis(),
+                                                            repositoryWorker, cache,
+                                                            encryptionStorageManager);
+        } catch (Throwable t) {
+            throw new StorageException("failed to create the repository while migrating. " +
+                                       "repositoryName: " + projectRepositoryName(repositoryName), t);
+        }
+
+        final Revision headRevision = oldRepository.normalizeNow(Revision.HEAD);
+        Revision baseRevision = null;
+        try {
+            for (int i = 2; i <= headRevision.major(); i++) {
+                baseRevision = new Revision(i - 1);
+                final Revision revision = new Revision(i);
+                final CompletableFuture<List<Commit>> historyFuture =
+                        oldRepository.history(revision, revision, "/**");
+                final CompletableFuture<Map<String, Change<?>>> changesFuture =
+                        oldRepository.diff(baseRevision, revision, "/**");
+                CompletableFuture.allOf(historyFuture, changesFuture).join();
+                final List<Commit> commits = historyFuture.join();
+                assert commits.size() == 1;
+                final Commit commit = commits.get(0);
+                encryptedRepository.commit(baseRevision, commit.when(), commit.author(), commit.summary(),
+                                           changesFuture.join().values()).join();
+            }
+        } catch (Throwable t) {
+            encryptedRepository.internalClose();
+            encryptionStorageManager.deleteRepositoryData(parent.name(), repositoryName);
+            throw new StorageException("failed to migrate the contents of the repository '" +
+                                       projectRepositoryName(repositoryName) + "' to an encrypted repository." +
+                                       " baseRevision: " + baseRevision, t);
+        }
+
+        // Simply create the placeholder file to indicate that this repository is encrypted.
+        // TODO(minwoox): remove the old content of the repository using a plugin.
+        try {
+            Files.createFile(Paths.get(oldRepository.repoDir().getPath(), ENCRYPTED_REPO_PLACEHOLDER_FILE));
+        } catch (IOException e) {
+            encryptedRepository.internalClose();
+            encryptionStorageManager.deleteRepositoryData(parent.name(), repositoryName);
+            throw new StorageException("failed to create the encrypted repository placeholder file at: " +
+                                       oldRepository.repoDir(), e);
+        }
+
+        if (!replaceChild(repositoryName, oldRepository, encryptedRepository)) {
```

Comment:
Note) I assume that when `replaceChild` is called, some requests that hold a reference to the repository directly may not work as expected.

e.g. https://github.com/line/centraldogma/blob/07c0b47f0868258c298ac3cc24b64e8751f13414/server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java#L320

Reply:
> I assume that when replaceChild is called, some requests that hold a reference to the repository directly may not work as expected.

They are closed with the exception when close is called:
https://github.com/line/centraldogma/pull/1150/files#diff-9e5e51bc31de64b1d588b42e388211e865e26b95c89aa5580a6864891ee4bb1fR111
https://github.com/line/centraldogma/pull/1150/files#diff-72d685fc105d04237696310ce037c22c9896a3cb78349854e847eb20426b61f2R176